pkg/ingester/client/timeseries.go

```diff
@@ -0,0 +1,13 @@
+package client
+
+// PreallocTimeseries is a TimeSeries which preallocs slices on Unmarshall.
+type PreallocTimeseries struct {
+	TimeSeries
+}
+
+// Unmarshal implements proto.Message.
+func (p *PreallocTimeseries) Unmarshal(dAtA []byte) error {
+	p.Labels = make([]LabelPair, 0, 20)
+	p.Samples = make([]Sample, 0, 10)
```

Comment:
Do we expect the call have more than one sample per series, usually?

Reply:
Hard to say - when I had this as `1` I saw some allocations, so I just stuck it at 10.